app/controllers/form_items_controller.rb

```diff
@@ -0,0 +1,21 @@
+class FormItemsController < ApplicationController
+  # authorization via cancan
+  load_and_authorize_resource
+
+  def update
+    attrs = params[:form_item]
+
+    if attrs[:ancestry].blank? or !@form_item.check_ancestry_integrity(attrs[:ancestry])
+      return render nothing: true, status: 422
+    end
+
+    @form_item.ancestry = attrs[:ancestry]
+    @form_item.rank = attrs[:rank]
+
+    if @form_item.save
+      render nothing: true, status: 204
+    else
+      render nothing: true, status: 422
```

Comment:
Might as well do `render json: @form_item.errors, status: 422` so that it will be a little easier to debug.